spark/src/test/java/org/apache/iceberg/spark/data/parquet/vectorized/TestParquetVectorizedReads.java

```diff
@@ -231,4 +232,33 @@ public void testVectorizedReadsWithReallocatedArrowBuffers() throws IOException
           return record;
         });
   }
+
+  @Test
+  public void testReadsForTypePromotedColumns() throws Exception {
+    Schema writeSchema = new Schema(
+            required(100, "id", Types.LongType.get()),
+            optional(101, "int_data", Types.IntegerType.get()),
+            optional(102, "float_data", Types.FloatType.get()),
+            optional(103, "decimal_data", Types.DecimalType.of(10, 5))
```

Comment:
Nit: indentation in this function it off. It should be 2 indents (4 spaces) for continuation indents.